### ISSUE TITLE
no-navigation-without-resolve: false positive for links with optional ResolvedPathname href

### ISSUE THREAD

**Issue by Stadly** [sveltejs/eslint-plugin-svelte#1525]
### Description

The `no-navigation-without-resolve` rule reports a false positive on `<a>` elements when `href` is typed as an optional `ResolvedPathname` (i.e. `ResolvedPathname | undefined`).

The rule correctly passes `allowNullish: true` when checking link attributes, intending to permit nullish href values. However, when the href expression is not itself nullish but has a *type* that includes `undefined` (because the property is optional), neither `expressionIsNullish` nor `expressionIsResolvedPathname` matches:

- `expressionIsNullish` checks whether the expression is syntactically nullish (e.g. a literal `undefined`), not whether the type is nullable — so it returns `false` for `href`
- `expressionIsResolvedPathname` performs a bidirectional assignability check against `ResolvedPathname`, which fails for `ResolvedPathname | undefined` because `undefined` is not assignable to `ResolvedPathname`

### Reproduction

```svelte
<script lang="ts">
import type { ResolvedPathname } from '$app/types';

interface Props {
  href?: ResolvedPathname;
}

const { href }: Props = $props();
</script>

<a {href}>Click me!</a>
```

**Expected:** No error — `href` is either `ResolvedPathname` or `undefined`, both of which are valid for `<a href>`.

**Actual:** `linkWithoutResolve` is reported.

### Suggested fix

Strip nullability from the node type in `expressionIsResolvedPathname` before performing the bidirectional assignability check, using `checker.getNonNullableType()`. This way, `ResolvedPathname | undefined` is reduced to `ResolvedPathname`, which correctly passes the check.

See PR #1523.

**Comment by marekdedic** [comment]
Thanks, I now understand the issue. This is the critical piece of code:

https://github.com/sveltejs/eslint-plugin-svelte/blob/03f7bd27fc7d042ce35f23c8b56dd626ddd1a380/packages/eslint-plugin-svelte/src/rules/no-navigation-without-resolve.ts#L336-L341

Your example doesn't match any of the `or` cases, because it's a `ResolvedPathname` and nullish *at the same time*. I can confirm this is really a bug in the rule. However, your PR is unfortunately too simple. I think we need to correctly handle the "control flow" in the above linked code. Also I need to think about other possible combinations :/

Thanks!

**Comment by marekdedic** [comment]
Hmm, it looks like `expressionIsNullish` doesn't even detect `null`/`undefined` types, just values - this triggers an error, but should be valid:

```svelte
<script lang="ts">
	interface Props {
		one: undefined;
		two: null;
		href: null;
	}

	const { one, two, href }: Props = $props();
</script>

<a href={one}>Click me!</a>
<a href={two}>Click me!</a>
<a {href}>Click me!</a>
```

**Comment by marekdedic** [comment]
In the end, I made my own PR (which, paradoxically, added some code only to remove it afterwards) that fixes your original issue _as well as_ the recognition of nullish types in general.

Thank you for spotting this!
